src/Microsoft.Data.Sqlite/SqliteCommand.cs

```diff
@@ -111,10 +112,10 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
             do
             {
                 Sqlite3StmtHandle stmt;
-                var rc = NativeMethods.sqlite3_prepare16_v2(
+                var rc = NativeMethods.sqlite3_prepare_v2(
                     Connection.DbHandle,
                     tail,
-                    -1,
+                    Encoding.UTF8.GetByteCount(tail) + 1, // add one to include the nul-terminator. See SQLite.org docs for sqlite3_prepare_v2.
```

Comment:
I'm not sure re-calculating this makes it worth it. :smile: The previous code set it lazily just after converting it to UTF8.